velox/exec/tests/TaskTest.cpp

```diff
@@ -581,4 +597,46 @@ TEST_F(TaskTest, singleThreadedCrossJoin) {
   }
 }
 
+// Check if a downstream task can elegantly close all related upstream tasks
+// once the downstream task completes/aborts.
+TEST_F(TaskTest, checkExchangeSourceClosedAfterAbort) {
+  // Create and start a "remote" task
+  std::string remoteTaskId = "mock://task-0";
+  auto remoteTaskPlan =
```

Comment:
I'd expect that this tests would orchestrate a scenario where the query gets stuck, i.e. the leaf task fills up the output buffer and keeps waiting for it to be cleared. It doesn't look like this test creates such conditions. Hence, it cannot be used to reproduce the original problem, debug it to identify root cause and then verify that the fix is effective.